ledger/sandbox/src/main/scala/com/digitalasset/platform/logging/ContextualizedLogger.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.logging
+
+import org.slf4j.{Logger, LoggerFactory}
+
+import scala.collection.concurrent.TrieMap
+import scala.reflect.ClassTag
+
+object ContextualizedLogger {
+
+  // Caches loggers to prevent them from needlessly waste memory
+  // Replicates the behavior of the underlying Slf4j logger factory
+  private[this] val cache = TrieMap.empty[String, ContextualizedLogger]
```

Comment:
You would create a `Logger` via the factory, and this ensures it's created only once because `LoggerFactory` caches loggers. This caches the logger wrapper as well. In most current usages I'd say we would end up creating few objects with this logger, but I'd prefer it to have the same behavior of `LoggerFactory` in case we start using it on smaller, more numerous objects.